test/integration/short/prepare-simulator-tests.js

```diff
@@ -0,0 +1,137 @@
+            });
+            client.connect.bind(client);
+            next();
+          },
+          function primeQuery(next) {
+            sCluster.primeQueryWithEmptyResult(query, next);
+          },
+          function clearLog(next) {
+            sCluster.clearLog(next);
+          }
+        ], done);
+    });
+    afterEach(function (done) {
+      client.shutdown.bind(client);
+      sCluster.destroy(done);
+    });
+    after(function (done) {
+      simulacron.stop(done);
+    });
+    it('should prepare query on all hosts', function (done) {
+      var idRandom = types.Uuid.random();
+      client.execute(query, [idRandom], {prepare: 1}, function (err, result) {
+        assert.ifError(err);
+        assert.strictEqual(client.hosts.length, 5);
+        assert.notEqual(result, null);
+        assert.notEqual(result.rows, null);
+        utils.eachSeries(client.hosts.values(), function(host, next) {
+          sCluster.queryNodeLog(host.address, function(logs) {
+            assert.ifError(err);
+            var prepareQuery = logs.find(function(queryLog) {
+              return queryLog.type === "PREPARE"
+                && queryLog.query == query;
+            });
+            assert.notEqual(prepareQuery, undefined);
+            next();
+          });
+        }, done);
+      });
+    });
+    it('should re-prepare query when host go UP again', function (done) {
+      var idRandom = types.Uuid.random();
+      var nodeDownAddress = sCluster.getContactPoints()[4];
+      utils.series(
+        [
+          function stopLastNode(next) {
+            sCluster.stopNode(nodeDownAddress, next);
+          },
+          function prepareQuery(next) {
+            client.execute(query, [idRandom], {prepare: 1}, function (err, result) {
+              assert.ifError(err);
+              assert.strictEqual(client.hosts.length, 5);
+              assert.notEqual(result, null);
+              assert.notEqual(result.rows, null);
+              next();
+            });
+          },
+          function verifyLogs(next) {
+            utils.eachSeries(client.hosts.values(), function(host, nextHost) {
+              sCluster.queryNodeLog(host.address, function(logs) {
+                var prepareQuery = logs.find(function(queryLog) {
+                  return queryLog.type === "PREPARE"
+                    && queryLog.query == query;
+                });
+                if (prepareQuery == undefined) {
+                  assert.strictEqual(nodeDownAddress, host.address);
+                } else {
+                  assert.notEqual(prepareQuery, undefined);
+                }
+                nextHost();
+              });
+            }, next);
+          },
+          function resumeLastNode(next) {
+            var timeout = null;
+            var nodeDown = client.hosts.get(nodeDownAddress);
+            nodeDown.on('up', function() {
+              clearTimeout(timeout);
+              helper.trace("Node marked as UP");
```

Comment:
On my local machine, the node is not marked as down and back up but the test succeeds.